File: frontend/src/types/transaction.ts
export interface Transaction {
  id: number;
  date: string;
  description: string;
  amount: number;
  category_id: number | null;
  statement_id: number;
  created_at: string;
}

export interface Statement {
  id: number;
  filename: string;
  uploaded_at: string;
  transaction_count: number;
  start_date: string | null;
  end_date: string | null;
}

export interface Category {
  id: number;
  name: string;
  keywords: string;
}

export interface CategoryTotal {
  category: string;
  total: number;
}

export interface MonthlyTotal {
  year: number;
  month: number;
  total: number;
}

File: backend/app/models/transaction.py
from sqlalchemy import Column, Integer, String, Numeric, Date, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from app.db import Base


class Transaction(Base):
    __tablename__ = "transactions"

    id = Column(Integer, primary_key=True, index=True)
    date = Column(Date, nullable=False)
    description = Column(String, nullable=False)
    amount = Column(Numeric(12, 2), nullable=False)  # negativo = saída, positivo = entrada
    category_id = Column(Integer, ForeignKey("categories.id"), nullable=True)
    statement_id = Column(
        Integer,
        ForeignKey("statements.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    category = relationship("Category", back_populates="transactions")
    statement = relationship("Statement", back_populates="transactions")
